include/generate/animation/grouptemplate.h

```diff
@@ -1,110 +1,225 @@
+                          ImportContainer& import,
+                          ID eventid,
+                          ID variableID,
+                          std::string masterFormat,
+                          NewAnimLock& atomicLock,
+                          int groupCount = 0);
+    void stateReplacer(std::string& line,
+                       std::string filename,
+                       std::string statenum,
+                       int ID,
+                       int stateID,
+                       int linecount,
+                       int groupMulti);
+    void processing(std::string& line,
+                    std::string filename,
+                    std::string masterFormat,
+                    int linecount,
+                    ID eventid,
+                    ID variableid,
+                    int groupMulti          = -1,
+                    int optionMulti         = -1,
+                    int animMulti           = -1,
+                    std::string multiOption = "");
+    void newID();
+    void setZeroEvent(std::string eventname);
+    void setZeroVariable(std::string variablename);
 };
 
 struct ExistingFunction : public JointTemplate
 {
 private:
-	bool m_hasGroup;
-	int* nextFunctionID;
-	ImportContainer* newImport;
-	std::string strID;
-	std::string format;
-	SSMap IDExist;
-	std::shared_ptr<master> subFunctionIDs;
-	std::vector<std::vector<std::shared_ptr<AnimationInfo>>> groupAnimInfo;
+    bool m_hasGroup;
+    int* nextFunctionID;
+    ImportContainer* newImport;
+    std::string strID;
+    std::string format;
+    SSMap IDExist;
+    std::shared_ptr<master> subFunctionIDs;
+    std::vector<std::vector<std::shared_ptr<AnimationInfo>>> groupAnimInfo;
 
 public:
-	vecstr groupExistingFunctionProcess(int curFunctionID, vecstr existingFunctionLines, std::shared_ptr<master> newSubFunctionIDs,
-		std::vector<std::vector<std::shared_ptr<AnimationInfo>>> newGroupAnimInfo, std::string format, ImportContainer& import, id newEventID, id newVariableID,
-		int& nFunctionID, bool hasMaster, bool hasGroup, setstr templateGroup, bool ignoreGroup);
-	void outPutExistingFunction(vecstr& existingFunctionLines, vecstr& newFunctionLines, bool isGroup, bool isMaster, bool ignoreGroup, std::string IDFileName,
-		std::vector<std::vector<std::unordered_map<std::string, bool>>>& masterOptionPicked, std::unordered_map<std::string, bool>& otherAnimType, int order, int groupOrder,
-		bool& isElement, int& elementLine, __int64& openRange, std::string& multiOption, int& elementCount, int curFunctionID, int curLine, unsigned int scopeSize);
-	void processing(std::string& line, std::string filename, int curFunctionID, int linecount, id eventid, id variableid, int groupMulti, int optionMulti = -1,
-		int animMulti = -1, std::string multiOption = "");
-	inline void newID();
-	void setZeroEvent(std::string eventname);
-	void setZeroVariable(std::string variablename);
-
+    VecStr
+    groupExistingFunctionProcess(int curFunctionID,
```

Comment:
why tho?

Reply:
Why what?

Reply:
I think Skikyo is referring to the return type being on a separate line.

Reply:
Oh, that. i don't know, that's clang format. I think it can be tweaked.

Reply:
yea i meant the return type being on a separate line. I have no problem with using Capital letter VecStr